### PR TITLE
Refactor Sliding Sync tests to better utilize the `SlidingSyncBase.do_sync(...)` (pt. 2)

### PR DESCRIPTION
**Pull request by MadLittleMods**
Refactor Sliding Sync tests to better utilize the `SlidingSyncBase.do_sync(...)` (pt. 2)

`SlidingSyncBase.do_sync()` for tests was first introduced in https://github.com/element-hq/synapse/pull/17452

Part 1: https://github.com/element-hq/synapse/pull/17481

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] [Code style](https://element-hq.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: changelog.d/17482.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Refactor Sliding Sync tests to better utilize the `SlidingSyncBase`.
```
Comment (MadLittleMods) [inline changelog.d/17482.misc:1]: Matches changelog from https://github.com/element-hq/synapse/pull/17481

File: tests/rest/client/test_sync.py [hunk 2]
```diff
@@ -1469,60 +1469,53 @@ def test_sync_list(self) -> None:
         """
         Test that room IDs show up in the Sliding Sync `lists`
         """
-        alice_user_id = self.register_user("alice", "correcthorse")
-        alice_access_token = self.login(alice_user_id, "correcthorse")
+        user1_id = self.register_user("user1", "pass")
+        user1_tok = self.login(user1_id, "pass")
```
Comment (MadLittleMods) [inline tests/rest/client/test_sync.py:1473]: This PR is focused on refactoring to use `do_sync(...)` but there are few other things I saw along the way to standardize with the rest of the tests.

For example, we use the `user1` pattern for the majority of the tests so I'm aligning this.

File: tests/rest/client/test_sync.py [hunk 3]
```diff
@@ -4807,59 +4563,47 @@ def test_data_incremental_sync(self) -> None:
...
-                        "enabled": True,
-                        "since": next_batch,
-                    }
-                },
+        sync_body = {
+            "lists": {},
+            "extensions": {
+                "to_device": {
+                    "enabled": True,
+                    "since": next_batch,
```
Comment (MadLittleMods) [inline tests/rest/client/test_sync.py:4571]: I see these [lints from CI](https://github.com/element-hq/synapse/actions/runs/10085765317/job/27887170453?pr=17482) but not locally:
```
tests/rest/client/test_sync.py:4571: error: Dict entry 1 has incompatible type "str": "str"; expected "str": "bool"  [dict-item]
tests/rest/client/test_sync.py:4588: error: Dict entry 1 has incompatible type "str": "str"; expected "str": "bool"  [dict-item]
```

Comment (MadLittleMods) [reply tests/rest/client/test_sync.py:4571]: Fixed with a `JsonDict` type :white_check_mark: 

I can see the error locally after updating my installed Synapse dependencies (`poetry install --extras all`)

File: tests/rest/client/test_sync.py [hunk 4]
```diff
@@ -1902,83 +1885,65 @@ def test_sliced_windows(self) -> None:
...
-            "POST",
-            self.sync_endpoint,
-            {
-                "lists": {
-                    "foo-list": {
-                        "ranges": [[0, 1]],
-                        "required_state": [
-                            ["m.room.join_rules", ""],
-                            ["m.room.history_visibility", ""],
-                            ["m.space.child", "*"],
```
Comment (erikjohnston) [inline tests/rest/client/test_sync.py:1955]: These have been removed, but I think its fine?

**Review by erikjohnston — APPROVED** [review]
woop

File: tests/rest/client/test_sync.py (hunk 4 above)
Comment (MadLittleMods) [reply tests/rest/client/test_sync.py:1955]: :+1: Removed because we don't need `required_state` for this test that looks at the list slices

**Comment by MadLittleMods** [comment]
Thanks for the review @erikjohnston 🐴

**Pull request merged**